tests/spec/s-object.js

```diff
@@ -356,4 +356,191 @@ describe('Object', function () {
+            expect(hasOwn.call(obj2, 'valueOf')).toBe(false);
+        });
+
+        it('not defined on object "hasOwnProperty"', function () {
+            expect(hasOwn.call(obj2, 'hasOwnProperty')).toBe(false);
+        });
+
+        it('not defined on object "isPrototypeOf"', function () {
+            expect(hasOwn.call(obj2, 'isPrototypeOf')).toBe(false);
+        });
+
+        it('not defined on object "propertyIsEnumerable"', function () {
+            expect(hasOwn.call(obj2, 'propertyIsEnumerable')).toBe(false);
+        });
+
+        it('not defined on object "constructor"', function () {
+            expect(hasOwn.call(obj2, 'constructor')).toBe(false);
+        });
+
+        it('not defined on object should be not ok in each case', function () {
+            expect(hasOwn.call(obj2, 'foo')).toBe(false);
+            expect(hasOwn.call(obj2, 'bar')).toBe(false);
+            expect(hasOwn.call(obj2, 'fuz')).toBe(false);
+        });
+
+        it('defined on object with "undefined" value "toString"', function () {
+            expect(hasOwn.call({
+                toString: undefined
+            }, 'toString')).toBe(true);
+        });
+
+        it('defined on object with "undefined" value "toLocaleString"', function () {
+            expect(hasOwn.call({
+                toLocaleString: undefined
+            }, 'toLocaleString')).toBe(true);
+        });
+
+        it('defined on object with "undefined" value "valueOf"', function () {
+            expect(hasOwn.call({
+                valueOf: undefined
+            }, 'valueOf')).toBe(true);
+        });
+
+        it('defined on object with "undefined" value "hasOwnProperty"', function () {
+            expect(hasOwn.call({
+                hasOwnProperty: undefined
+            }, 'hasOwnProperty')).toBe(true);
+        });
+
+        it('defined on object with "undefined" value "isPrototypeOf"', function () {
+            expect(hasOwn.call({
+                isPrototypeOf: undefined
+            }, 'isPrototypeOf')).toBe(true);
+        });
+
+        it('defined on object with "undefined" value "propertyIsEnumerable"', function () {
+            expect(hasOwn.call({
+                propertyIsEnumerable: undefined
+            }, 'propertyIsEnumerable')).toBe(true);
+        });
+
+        it('defined on object with "undefined" value "constructor"', function () {
+            expect({
+                constructor: undefined
+            }.hasOwnProperty('constructor')).toBe(true);
+        });
+
+        it('string defined', function () {
+            var str = 'abc';
+            expect(str.hasOwnProperty('0')).toBe(true);
+            expect(str.hasOwnProperty('1')).toBe(true);
+            expect(str.hasOwnProperty('2')).toBe(true);
+        });
+
+        it('string not-defined', function () {
+            var str = 'abc';
+            expect(str.hasOwnProperty('3')).toBe(false);
+        });
+
+        it('string object defined', function () {
```

Comment:
similarly, patching this would require replacing `Object`, and would give a false report if someone was using this to determine if bracket access on strings is safe